pkg/api/controllers/log/websocket.go

```diff
@@ -160,10 +161,20 @@ func ReadServerLog(ginCtx *gin.Context) {
 
 func ReadWorkspaceLog(ginCtx *gin.Context) {
 	workspaceId := ginCtx.Param("workspaceId")
-
 	server := server.GetInstance(nil)
 
-	wsLogReader, err := server.WorkspaceService.GetWorkspaceLogReader(workspaceId)
+	var wsLogReader io.Reader
+	var err error
+
+	maxRetries := 5
```

Comment:
I think this should be a query parameter of the request. If the parameter is not set, you can retry indefinitely.